src/wallet/wallet.h

```diff
@@ -593,6 +594,15 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
     MasterKeyMap mapMasterKeys;
     unsigned int nMasterKeyMaxID;
 
+    /**
+     * Used to keep track of unspent outpoints
+     */
+    typedef boost::unordered_map<COutPoint, std::pair<const CWalletTx*, isminetype> > UnspentsMap;
```

Comment:
Should we keep also a `boost::unordered_map<CWalletTx*, COutPoint>` to better filter invalid coins?